src/state/slices/opportunitiesSlice/resolvers/types.ts

```diff
@@ -0,0 +1,20 @@
+import type { BaseQueryApi } from '@reduxjs/toolkit/dist/query/baseQueryTypes'
+import type { DefiProvider, DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+
+import type { foxFarmingLpMetadataResolver, foxFarmingLpUserDataResolver } from './foxfarming'
+
+export type ReduxApi = Pick<BaseQueryApi, 'dispatch' | 'getState'>
+
+export type IDefiProviderToMetadataResolverByDeFiType = {
+  [key in DefiProvider]?: {
+    // TODO: Discriminated union, only FOX ETH LP/Farming is implemented so far
+    [key in DefiType]: typeof foxFarmingLpMetadataResolver
+  }
+}
+
+export type IDefiProviderToUserDataResolverByDeFiType = {
+  [key in DefiProvider]?: {
+    // TODO: Discriminated union, only FOX ETH LP/Farming is implemented so far
+    [key in DefiType]: typeof foxFarmingLpUserDataResolver
+  }
+}
```

Comment:
Since these dynamic keys are just Enums, we can use utility types, which feels a little cleaner:

```suggestion
export type IDefiProviderToMetadataResolverByDeFiType = Partial<
  Record<DefiProviders, Record<DefiType, typeof foxFarmingLpMetadataResolver>>
>

export type IDefiProviderToUserDataResolverByDeFiType = Partial<
  Record<DefiProviders, Record<DefiType, typeof foxFarmingLpUserDataResolver>>
>
```